WEB/Src/WindowsServer/WindowsServer/Implementation/IAzureMetadataRequestor.cs

```diff
@@ -1,10 +1,11 @@
 ﻿namespace Microsoft.ApplicationInsights.WindowsServer.Implementation
 {
+    using System;
     using System.Collections.Generic;
     using System.Threading.Tasks;
     using Microsoft.ApplicationInsights.WindowsServer.Implementation.DataContracts;
 
-    internal interface IAzureMetadataRequestor
+    internal interface IAzureMetadataRequestor : IDisposable
```

Comment:
Do we need this?

Reply:
The call stack here is: TelemetryModule > Provider > Requestor > HttpClient.

We don't NEED this interface to be disposable. 
Alternatively, the Provider can check if it's private requestor implements IDisposable.
This introduces a little more complexity and it's not as obvious that the Requestor needs to be disposed. (The Analyzers can't catch this scenario).

Reply:
If we don't need this, let's try not to add it.

Reply:
FYI this might be a good approach https://github.com/open-telemetry/opentelemetry-dotnet/blob/95893457080a772eb75e7885e640d2c055b2a0e1/src/OpenTelemetry/Trace/TracerProviderSdk.cs#L304